app/indexers/hyrax/indexers/pcdm_collection_indexer.rb

```diff
@@ -9,7 +9,8 @@ class PcdmCollectionIndexer < Hyrax::Indexers::ResourceIndexer
       include Hyrax::VisibilityIndexer
       include Hyrax::LocationIndexer
       include Hyrax::ThumbnailIndexer
-      include Hyrax::Indexer(:core_metadata)
+      include Hyrax::Indexer(:core_metadata) unless Hyrax.config.flexible?
+      include Hyrax::Indexer('Hyrax::PcdmCollection') if Hyrax.config.flexible?
```

Comment:
I want to push back lightly against the expectation that applications are using `Hyrax::PcdmCollection` as their collection model (and not something which inherits from `Hyrax::PcdmCollection`). The collection model is configurable in Hyrax, and it’s a bit strange to expect flexible metadata to be defined for `Hyrax::PcdmCollection` when the collection model being used is, in fact, `class ::Collection < Hyrax::PcdmCollection`. ([This is in fact the case in Koppie.](https://github.com/samvera/hyrax/blob/main/.koppie/app/models/collection_resource.rb))

Reply:
This doesn't prevent inheritance. We also inherit from PcdmCollection in Hyku. :) But it does require that the m3 profile include PcdmCollection, so we are removing line 13. Since this model only includes core metadata, there's no need to support flexible metadata here as an equivalence.

Reply:
I’m not concerned about inheritance behaviours so much as the experience for application developers / end‐users. If my collection class is named `MyCollection`, I would expect to be able to add metadata properties to `MyCollection` in the M3 and have it work. However, that isn’t the case here; only `Hyrax::PcdmCollection` adds collection properties, and there is no mechanism for loading collection metadata by a different name.

Reply:
The collection work is still not finalized. I honestly thought we had already included it in the dassie and koppie models but it appears that isn't the case. Thanks for pointing that out. You're right that it should be based on your resource class name.